docs/glossary.rst

```diff
@@ -61,4 +61,14 @@ Let's go through a few words and their meaning.
         are namespaces for objects of the database. See the `PostgreSQL Schema
         documentation`_ for more detail.
 
+    Lock
+        When configuring a job using `Task.configure` you can attach a lock to the job.
+        Jobs with the same lock are guaranteed to be executed in defer order, and in a
+        sequential manner. No two jobs with the same lock can run simultaneously. See
+        the `howto-locks` section for more information.
+
+    Queueing Lock
+        When configuring a job using `Task.configure` you can attach a queueing lock to
+        the job. No two jobs with the same queueing lock can be waiting in the queue.
```

Comment:
Shouldn't we talk about `AlreadyEnqueued` here? Otherwise what happens if we try to enqueue two jobs is left unclear.

Reply:
@ewjoachim said otherwise so I removed it with https://github.com/peopledoc/procrastinate/pull/219/commits/98a068a0b4fa2a2401227ac4b8e059aec4351910. But happy to add it back if others think it's worth mentioning `AlreadyEnqueued` here.

Reply:
I think the glossary should be there to explain the terms, not the behaviour. That being said, we can link to pages that explain this. For this feature, I think there is no dedicated page. Maybe we could link to the ref doc...

Reply:
Ok, makes sense :+1: . I think I was biased because this was the first thing I read from this PR, but this will probably not be the typical user experience :) .